Update example and Readme for otlp log exporter (#4747)
open-telemetry/opentelemetry-dotnet: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md

* `ExportProcessorType`: Whether the exporter should use [Batch or Simple
  exporting
  processor](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#built-in-span-processors).

>   The default is Batch. **Note**: This option is only applicable to

Reviewer: Add a section for `LogRecordExportProcessorOptions` as well?

Author: May need a bigger refactoring for this. This also include `MetricReaderOptions` part. Would like to keep it in separate PR if that is ok. For immediate usage, the link to testlogs.cs should work.

>   The default is Batch. **Note**: This option is only applicable in case of

Reviewer: ```suggestion
  The default is Batch. **Note**: This option is only applicable to
```

>   The default is Batch. **Note**: This option is only applicable in case of

Reviewer: You could probably move these two properties to a sub-heading such as "Traces specific properties"

Author: This is just temporary. We can remove this completely if we do https://github.com/open-telemetry/opentelemetry-dotnet/issues/4527

Author: updated

  trace exporter.

* `BatchExportProcessorOptions`: Configuration options for the batch exporter.
  Only used if ExportProcessorType is set to Batch. **Note**: This option is

>   only applicable in case of trace exporter.

Reviewer: ```suggestion
  only applicable to trace exporter.
```

* `TimeoutMilliseconds` : Max waiting time for the backend to process a batch.

> The following options are only applicable to trace exporter:

Reviewer: ```suggestion
The following options are only applicable to `OtlpTraceExporter`:
```

| `OTEL_EXPORTER_OTLP_TIMEOUT`  | `TimeoutMilliseconds`                 |
| `OTEL_EXPORTER_OTLP_PROTOCOL` | `Protocol` (`grpc` or `http/protobuf`)|

The following environment variables can be used to override the default values

> for `BatchExportProcessorOptions` in case of `trace` exporter (following the

Reviewer: ```suggestion
for `BatchExportProcessorOptions` in case of `OtlpTraceExporter` (following the
```

[OpenTelemetry
specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#batch-span-processor))

> | Environment variable             | `BatchExportProcessorOptions` property  |

Reviewer: ```suggestion
| Environment variable             | `OtlpExporterOptions.BatchExportProcessorOptions` property  |
```

| ---------------------------------| ----------------------------------------|
| `OTEL_BSP_SCHEDULE_DELAY`        | `ScheduledDelayMilliseconds`            |
| `OTEL_BSP_EXPORT_TIMEOUT`        | `ExporterTimeoutMilliseconds`           |
| `OTEL_BSP_MAX_QUEUE_SIZE`        | `MaxQueueSize`                          |
| `OTEL_BSP_MAX_EXPORT_BATCH_SIZE` | `MaxExportBatchSize`                    |

The following environment variables can be used to override the default values

> for `BatchExportProcessorOptions` in case of `log` exporter (following the

Reviewer: ```suggestion
for `BatchExportProcessorOptions` in case of `OtlpLogExporter` (following the
```

[OpenTelemetry
specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#batch-logrecord-processor))

> | Environment variable              | `BatchExportProcessorOptions` property  |

Reviewer: ```suggestion
| Environment variable              | `LogRecordExportProcessorOptions.BatchExportProcessorOptions` property  |
```
